### ISSUE TITLE
Ability to return to the GDM from various locations

### ISSUE THREAD

**Issue by selinad** [ClinGen/clincoded#326]
Right now, we lack the ability return to the GDM from most places - this can be frustrating - would be great to have link that takes one back from various places.  Could there be a button to get back to current paper in GDM as well?

**Comment by selinad** [comment]
If possible, we think this would be high value for R3

**Comment by mrmin123** [comment]
So I have this implemented, but the visuals need work:

On edit pages:
![image](https://cloud.githubusercontent.com/assets/4326866/12247919/13909cb4-b86b-11e5-977c-1b94c7bc7990.png)

On view pages:
![image](https://cloud.githubusercontent.com/assets/4326866/12247925/1fafc948-b86b-11e5-8cd9-a0e956e6b2b7.png)

The link is disabled where it shouldn't link to the GDM:
On curation central:
![image](https://cloud.githubusercontent.com/assets/4326866/12247906/0a3e273a-b86b-11e5-90d8-5401244d4c6f.png)

On post-submit page:
![image](https://cloud.githubusercontent.com/assets/4326866/12247942/3d0c6cb2-b86b-11e5-9983-a640888bc055.png)


Assuming the non-linked header remains black w/ no underline, how should the linked header display? Black w/ underline? Blue-ish w/o underline until hover? Maybe leave the text black w/ no underline, but display a 'link' icon next to it?

**Comment by selinad** [comment]
Cool !  I'm wondering if there should be an icon next to the record name instead of linking the record name itself?

**Comment by mrmin123** [comment]
Definitely a possibility...
Some 'typical' link icons would be things like these:

* http://fortawesome.github.io/Font-Awesome/icon/link/
* http://fortawesome.github.io/Font-Awesome/icon/external-link/
* http://fortawesome.github.io/Font-Awesome/icon/external-link-square/

Although we could theoretically use anything from font awesome (depending on version): http://fortawesome.github.io/Font-Awesome/icons/

**Comment by kilodalton** [comment]
I would avoid using one with an arrow as that often implies sharing with or outputting to another resource (Twitter, etc)

**Comment by selinad** [comment]
A couple of other ideas...may be too far-fetched:

http://fortawesome.github.io/Font-Awesome/icon/briefcase/
http://fortawesome.github.io/Font-Awesome/icon/folder/
http://fortawesome.github.io/Font-Awesome/icon/folder-open/
http://fortawesome.github.io/Font-Awesome/icon/bookmark/
http://fortawesome.github.io/Font-Awesome/icon/compass/
http://fortawesome.github.io/Font-Awesome/icon/anchor/
http://fortawesome.github.io/Font-Awesome/icon/level-up/
http://fortawesome.github.io/Font-Awesome/icon/th-large/

**Comment by selinad** [comment]
It could go up next to the home button (maybe?)

**Comment by mrmin123** [comment]
Icon next to header text:
![image](https://cloud.githubusercontent.com/assets/4326866/12280922/9a2ad43e-b946-11e5-9792-e96e47de1dcc.png)

**Comment by wrightmw** [comment]
I kind of like the link where it is next to the GDM itself since the link takes you to the homepage for that GDM. Since it's a link to the GDM homepage, how about http://fortawesome.github.io/Font-Awesome/icon/home/?

**Comment by wrightmw** [comment]
Although maybe a 'home' button would be misconstrued as a link to the main homepage... I'm already arguing against my own idea, is that a sign of a split personality? ;-)

**Comment by mrmin123** [comment]
I feel like it's a bit weird to have two icons that do different things/go different places have the same icon, but that could just be me:

![image](https://cloud.githubusercontent.com/assets/4326866/12307880/e5e54caa-b9f6-11e5-955a-e69f5d82a4fa.png)

**Comment by forresttanaka** [comment]
It’s not just you. I think that’s really confusing. A couple suggestions off the top of my head:

1. Establish a symbolic “brand” for each kind of page, like the briefcase for GDM maybe. Then you can link to that symbol.
2. Turn the GDM spec in the gray header into an obvious rounded-corner button. This button would be huge sometimes, but I don’t think anyone would have any question about where it goes.
3. No symbol at all: just a button saying something like “Gene-Disease.” Might not be obvious what it means at first, but it would after the first time they clicked it.
4. Use breadcrumbs on every page. The top level breadcrumb would be for the GDM home.

**Comment by selinad** [comment]
I like 1 and 4!  One of the font awesome icons I had suggested was the briefcase...

**Comment by mrmin123** [comment]
![image](https://cloud.githubusercontent.com/assets/4326866/12309104/d65c3f26-b9fd-11e5-97a3-fb2fca2648db.png)

Need to work on a mockup for the breadcrumbs...

**Comment by forresttanaka** [comment]
I made a breadcrumb module for React if you want to pillage.

**Comment by mrmin123** [comment]
@forresttanaka Could you link me to it?

**Comment by forresttanaka** [comment]
Bottom of [this file](https://github.com/ENCODE-DCC/encoded/blob/master/src/encoded/static/components/navbar.js). Just remembered it might need to work very differently for ClinGen. ENCODE has a very specific way to do breadcrumbs.

**Comment by mrmin123** [comment]
Actually, couldn't we just repurpose these links as the breadcrumbs?:

![image](https://cloud.githubusercontent.com/assets/4326866/12309491/e9e3a80c-b9ff-11e5-82f9-0ee641a39791.png)

So it'd be something like [GDM link] // [F2] // i3

**Comment by forresttanaka** [comment]
Sounds like a good idea actually, and makes for no new part of the page for people to learn — just new functionality in what they know.

**Comment by mrmin123** [comment]
Right. I also wasn't sure where to place the breadcrumbs, but this seems like a decent place, or maybe even above the 'Edit Individual Information' header.

**Comment by selinad** [comment]
nice idea @mrmin123.  Would we also have icon?  Maybe the briefcase next to the home button and possibly next to the GDM name spelled out?  

Would the breadcrumbs cover all pages?

**Comment by mrmin123** [comment]
Icon:
![image](https://cloud.githubusercontent.com/assets/4326866/12309804/756ac2ec-ba01-11e5-84b4-e747d05983c2.png)

Breadcrumbs:
![image](https://cloud.githubusercontent.com/assets/4326866/12310189/bf4ba8a2-ba03-11e5-86ba-c5e10637e9e4.png)

@selinad: The breadcrumbs can be on the View and Edit pages. The main GDM and Submit pages would not have them by design.

**Comment by mrmin123** [comment]
I'm not sure about having the briefcase icon next to both the Home button and GDM name. They're so close together to begin with that it doesn't seem very useful to have them in both positions.

**Comment by selinad** [comment]
thx @mrmin123 - the Submit pages have a return to Record Curation Central button (or similar wording).  

I agree having a briefcase both places might be a bit much - maybe just next to the GDM text and then maybe next to the GDM breadcrumb (we can't call it GDM there), and then on the button for return to GDM on submit pages?

**Comment by mrmin123** [comment]
Something like this?:

Edit page:
![image](https://cloud.githubusercontent.com/assets/4326866/12312364/ab4f42b8-ba13-11e5-9b5d-23a8dbf4bc5a.png)

Submit page:
![image](https://cloud.githubusercontent.com/assets/4326866/12312382/d4b1ac72-ba13-11e5-9773-ebe83f10dd4d.png)

Should the View pages have both the partial header w/ briefcase link AND the breadcrumbs?

**Comment by selinad** [comment]
Nice @mrmin123 !

Does the briefcase at top need to be in a box?
Should briefcase go after GDM name and "Return to Record Curation page"?  What think you @wrightmw ?

**Comment by mrmin123** [comment]
Doesn't need to be in a box:
![image](https://cloud.githubusercontent.com/assets/4326866/12312958/8d1baca0-ba18-11e5-8ea2-44f22e10ab5e.png)

Can even be a bit larger:
![image](https://cloud.githubusercontent.com/assets/4326866/12312969/a11de556-ba18-11e5-99ef-7b08e359c8ce.png)

**Comment by selinad** [comment]
Nice!  I think I might prefer it aligned with top row...

**Comment by wrightmw** [comment]
Very nice! I agree with Selina, I think it would look better if the briefcase was aligned with the top row of text.

**Comment by mrmin123** [comment]
What I have for Individual edit page:
![image](https://cloud.githubusercontent.com/assets/4326866/12339621/d5a02962-baca-11e5-8530-9193151405f8.png)

What I have for Experimental edit page:
![image](https://cloud.githubusercontent.com/assets/4326866/12339638/e7ff7d24-baca-11e5-9d42-00d1c24cab73.png)

We also want the breadcrumbs in the View pages, correct?

**Comment by selinad** [comment]
Nice!

Since the View pages pop up a new window right now, I'm not certain we should implement them there...others?

**Comment by mrmin123** [comment]
Wasn't the plan to eventually make the links not open in a new window?

**Comment by selinad** [comment]
I'd like that...just saying until we change it we might not want to add breadcrumbs?

**Comment by mrmin123** [comment]
OK, works for me. There's still a link to the GDM on the view pages, unless you'd like that removed as well:

![image](https://cloud.githubusercontent.com/assets/4326866/12342042/5b9ad610-badb-11e5-86f8-b78cc41e00a9.png)

**Comment by selinad** [comment]
Maybe we should all discuss whether it should not open new tab now that the navigation functionality exists?

**Comment by wrightmw** [comment]
My preference would be that clicking on the briefcase takes you to that GDM homepage without opening a new tab.

**Comment by mrmin123** [comment]
I'm of the opinion that the user should have the option of opening links in new tabs/windows, and that the website itself should never force that option on the user, so my vote goes to no new tabs.

**Comment by wrightmw** [comment]
I'm also in favour tabs/windows as an option but not the default

**Comment by selinad** [comment]
me three! :+1:

**Comment by selinad** [comment]
There are a few things - like Kang's link to view free text pheno - is there option to open small browser window (more like a pop-up)?  It would seem weird to go off a page that you are entering info into...

**Comment by mrmin123** [comment]
OK, during our last meeting we discussed the possibility of making the 'View' links for the different elements popup modals. This is a bit more involved so for now I'm going to work on incorporating breadcrumbs into the View pages as well, and circle back to the modal option in the future.

**Comment by selinad** [comment]
Hi @mrmin123 - I'm not certain we should make breadcrumbs on View pages yet?  It seems it could lead to some strange navigation (probably OK, but it would keep taking one away from original page and seems like an awkward way to navigate through the site).

Others, please chime in if you disagree...

**Comment by wrightmw** [comment]
When Erin visits we can discuss her wish list for making the interface more user friendly.  Until then I agree with Selina that we should just implement the GDM links.

**Comment by selinad** [comment]
Hi @wrightmw - I'm thinking the breadcrumbs can go elsewhere, tho, just not View pages...I think they would be handy there...what do you think?

**Comment by mrmin123** [comment]
Sure. I guess they're technically more of a hierarchy view than a navigation breadcrumb, but I see your point of it being potentially confusing. Do you still want the link to the parent GDM in the View pages?

**Comment by wrightmw** [comment]
@selinad I'm not against breadcrumbs, I like them (although with my celiac disease I can't eat them)... I'm in agreement with adding breadcrumbs for hierarchy, just not in the View pages.

**Comment by mrmin123** [comment]
Sorry for the really long comment, but this is what I have set up right now. I can spin up an instance if you'd like to see it in action:

#### Curate/Edit pages:

Changes:

* Added clickable briefcase icon in GDM header that links back to GDM
* Added hierarchy breadcrumb below the Edit/Curate header

Group:
![image](https://cloud.githubusercontent.com/assets/4326866/12465173/9513a7c8-bf81-11e5-8415-36cbce56361d.png)

Family (under Group):
![image](https://cloud.githubusercontent.com/assets/4326866/12465187/a75bdd60-bf81-11e5-88ba-7a7320c3290a.png)

Individual (under Group):
![image](https://cloud.githubusercontent.com/assets/4326866/12465204/c00cca9a-bf81-11e5-960c-0e03a22f4ab5.png)

Individual (Under Group):
![image](https://cloud.githubusercontent.com/assets/4326866/12465214/ccc2da36-bf81-11e5-941d-2cb519e108ab.png)

Experimental Data:
![image](https://cloud.githubusercontent.com/assets/4326866/12465237/ecf5847a-bf81-11e5-8afc-fb02fa49baf3.png)

Variant (no breadcrumb):
![image](https://cloud.githubusercontent.com/assets/4326866/12465306/38022be4-bf82-11e5-99af-cc81f578a7f9.png)

#### Submit pages:

Changes:

* Added briefcase icon next to 'Return to Curation Central' text in buttons

Group submit:
![image](https://cloud.githubusercontent.com/assets/4326866/12465363/82b23882-bf82-11e5-9f35-914aa78f62ad.png)

Family submit:
![image](https://cloud.githubusercontent.com/assets/4326866/12465336/635e8b66-bf82-11e5-8aa8-eb3e210ba006.png)

Individual submit:
![image](https://cloud.githubusercontent.com/assets/4326866/12465348/72b6d5aa-bf82-11e5-9f78-27a8561e806a.png)

Experimental Data submit:
![image](https://cloud.githubusercontent.com/assets/4326866/12465380/91b21d02-bf82-11e5-9405-c2f66eb2ba5b.png)

#### View pages:

Changes:

* Added truncated GDM header, with a linked briefcase GDM icon

Example Group view:
![image](https://cloud.githubusercontent.com/assets/4326866/12465424/c8410dc4-bf82-11e5-86d8-11f49a8208d4.png)

**Comment by mrmin123** [comment]
@selinad: Continuing the convo in the chat... something like this?

![image](https://cloud.githubusercontent.com/assets/4326866/12466876/7b66827c-bf8c-11e5-9802-1a7b0e45d057.png)

**Comment by selinad** [comment]
I like it - thanks @mrmin123 ! Will wait for approval from @wrightmw 

One thing to consider moving forward - external link icon for variant (goes to ClinVar) and Mutalyzer, etc.  Otherwise, it looks like clicking on it will get one to ... well, the same page they are on. 

I don't think we have that anywhere yet, do we?

**Comment by mrmin123** [comment]
We don't have that anywhere yet, but we could use [this](http://fortawesome.github.io/Font-Awesome/icon/external-link/) or [this](http://fortawesome.github.io/Font-Awesome/icon/external-link-square/) icon since they're universally used as the 'external link' icon.

**Comment by selinad** [comment]
Thanks @mrmin123 - I have a preference for the first one, probably because it is most familiar to me - anyone else have opinions?

**Comment by wrightmw** [comment]
@mrmin123 I like it! Great work! I have a preference for the first icon, it seems more intuitive to me with the arrow leaving the box that it's an external link...

**Comment by mrmin123** [comment]
OK. Either way, the external link icon should be a separate ticket.

I've spun up a demo instance with the breadcrumbs/GDM linkbacks here: http://326-mc-gdm-linkbacks-breadcrumbs-f4de2c7-minchoi.instance.clinicalgenome.org/

GDM I created with entries are here: http://326-mc-gdm-linkbacks-breadcrumbs-f4de2c7-minchoi.instance.clinicalgenome.org/curation-central/?gdm=f43d2f9b-4614-43da-adae-3fcb223682c7&pmid=123456

**Comment by wrightmw** [comment]
In the instances I have run there is a problem with the 'Generate New Summary' button on the homepage. See attached screenshot... the 'Generate New Summary' button has blue text on a blue background. 
![gnsbutton](https://cloud.githubusercontent.com/assets/15131169/12662328/8b033772-c5d4-11e5-9657-42b66178e907.png)

**Comment by wrightmw** [comment]
The GDM home button worked for me from all pages.
When editing/adding new Groups/Families/Individuals the breadcrumbs work well.

**Comment by selinad** [comment]
Question - would it be strange to have a black or grayed out briefcase (non-clickable) next to the GDM on the GDM page?  (to help train)

**Comment by selinad** [comment]
Another question - would it make sense to remove the text for the GDM and just have the briefcase in the breadcrumb?
![screen shot 2016-01-28 at 4 08 34 pm](https://cloud.githubusercontent.com/assets/11320314/12663053/7dff318e-c5d9-11e5-9ef0-b293fce77a11.png)

Right now, there are 2 links (GDM name and briefcase) and the GDM is really long, plus written directly above.

**Comment by wrightmw** [comment]
@selinad Blimey, it does look odd when the GDM name is really long. I agree with you, I think just a briefcase would look better.

**Comment by selinad** [comment]
Apologies, @mrmin123 - I know we decided not to do breadcrumbs for the View page and now I feel like we are halfway consistent (my fault).  For instance, GRP1 is a link below, but we were trying to avoid links (which is why I discouraged breadcrumbs) and now there is a mix of links and no links on the View pages.  

@wrightmw - are you bothered by this?  It almost seems like we should have no links (force them to close tab) or same links and next step be to get rid of the tab behavior....

![screen shot 2016-01-28 at 4 10 47 pm](https://cloud.githubusercontent.com/assets/11320314/12663094/c34e5b0c-c5d9-11e5-92af-6db50193054d.png)

**Comment by mrmin123** [comment]
@selinad @wrightmw I will work on adding an unlinked briefcase to the header when applicable, and making the GDM link in the breadcrumbs just the briefcase. I'll also incorporate what you guys want for the View pages once you reach a consensus here :)

**Comment by selinad** [comment]
OK - following spontaneous meeting on topic:

* View pages will no longer open a new tab
* View option will be removed from Submit pages (Edit will remain)
* Breadcrumbs added to all View pages
* briefcase grayed/blacked out on GDM
* GDM name removed from breadcrumb - briefcase remains
* View header will remain as is now
* @wrightmw will spend the evening looking for external icons that open new tab or window :runner: 

@wrightmw and @mrmin123 - additions?

**Comment by mrmin123** [comment]
My one question: remaining Edit button on Submit pages will be renamed? 'View/Edit'? 'Edit/View'? Or remain as 'Edit'?

**Comment by selinad** [comment]
What would you think of keeping it as is on GDM page - Edit for Group and Edit/Assess for Family and Individual?  ( @wrightmw ?)

**Comment by wrightmw** [comment]
@selinad I already concede defeat. I have looked at 8 genomic resources and only 1 of them opened a new tab for an external link.  No boxing gloves required, on this issue you are the mightiest *ight...

**Comment by wrightmw** [comment]
I agree.. for consistency... 'Edit' for Group... 'Edit/Assess' for Family and Individual

**Comment by selinad** [comment]
Such a gentleman, even in defeat. :smile: 

You just forgot to give me my trophy :trophy:  :wink:

**Comment by wrightmw** [comment]
Here you go :trophy: :balloon: :gift:

**Comment by selinad** [comment]
Wow - little gold packages are always good, too!

**Comment by selinad** [comment]
@mrmin123  - here's a tough one (GRP1 link) - could it open a small pop-up?  Or, the user controls...
![screen shot 2016-01-28 at 5 28 21 pm](https://cloud.githubusercontent.com/assets/11320314/12664569/95327a9e-c5e5-11e5-919a-618c46d05975.png)

**Comment by selinad** [comment]
Another potential problem - adding assessment to a View page - upon submit, should now go to where one came from when clicking on view since new window not opened....

**Comment by mrmin123** [comment]
Do we want the same behavior for Variant assessments?

**Comment by mrmin123** [comment]
@selinad @wrightmw I believe this branch has everything we're looking for... I hope: http://326-mc-gdm-linkbacks-breadcrumbs-0a30d12-minchoi.instance.clinicalgenome.org/

**Comment by wrightmw** [comment]
Thanks @mrmin123 
@selinad When I enter a family with variants... I am given the option of Edit/Assess but really I can only edit the entry when I click on the button... do you think we should re-word the button?
![fff2](https://cloud.githubusercontent.com/assets/15131169/12692617/70ebad9e-c6ae-11e5-8f5f-7fd21269b7d9.png)

**Comment by wrightmw** [comment]
@mrmin123 When I'm in an edit individual page and I follow the instructions in the yellow box to create a proband... in this instance when I click on the 'Edit FFF1' link it goes to an empty page (see below):
![fff1](https://cloud.githubusercontent.com/assets/15131169/12726249/77a253a8-c8cb-11e5-83e0-53d5a3350cda.png)
![fff2](https://cloud.githubusercontent.com/assets/15131169/12726254/7f49aae8-c8cb-11e5-9059-74dcebc37602.png)

**Comment by wrightmw** [comment]
@mrmin123 I'm still testing this, but I'm very much enjoying how much easier the curation experience is in your instance. Traversing the curation interface is much more user-friendly. Thank you.

**Comment by wrightmw** [comment]
@mrmin123 After I've added an experiment, the briefcase on the title header is not clickable. I understand that I can click on the 'Return to Record Curation page' button but I think it would be good if the briefcase remained a link in all pages where we are ok with the curator linking through to the Record curation page. @selinad do you agree? See:
![expo5](https://cloud.githubusercontent.com/assets/15131169/12726819/7b03d334-c8ce-11e5-95a5-63efa051f915.png)

**Comment by mrmin123** [comment]
@wrightmw Glad to hear that you're finding navigation easier with this addition.

* I've fixed the Edit Family link not working from the Edit Individual curation page
* I can change it so that the Briefcase in the header for the Individual and Experimental Data submit pages if desired

**Comment by wrightmw** [comment]
Thanks @mrmin123

**Comment by selinad** [comment]
I agree with @wrightmw that it should be clickable - I think on all pages except the GDM itself.  I also see this problem after a Group Submit...

![screen shot 2016-02-01 at 10 42 33 am](https://cloud.githubusercontent.com/assets/11320314/12727177/8f5ea6ae-c8d0-11e5-804a-67588c6e4820.png)

**Comment by mrmin123** [comment]
@selinad The danger with having the briefcase be clickable in the Family and Group submit pages is that they can use it to (more easily) bypass the questions they have to answer in the submit page to continue.

**Comment by selinad** [comment]
Yes, that is true.  But it is far enough away that they would have to have decided to make that choice...so I think it's OK, but will let @wrightmw weigh in...

**Comment by selinad** [comment]
Note that the briefcase is clickable on a page where a user enters info (i.e. creating a Family) - this loses any entered data, which is also a way in which the user is note protected.  I think it's OK, though - my inclination is that users take a risk when they go away from a page where they have entered but not submitted data..

**Comment by selinad** [comment]
When we can, for a View Family page, we should list associated Individuals on that page - not sure if the Individual would be part of the breadcrumb, but there could be Associated Individual with a link to the Individual page

**Comment by mrmin123** [comment]
Gotcha. I'm not opposed to adding the links if you're OK with the user being able to bypass those questions on the submit page: the links that become available when answering the questions are accessible from the curation central anyway.

I'll wait for a consensus from the other *ight before implementing.

**Comment by selinad** [comment]
Thanks @mrmin123 

The navigation functionality is great!  I agree with @wrightmw !

One more thing - clicking on Experiment 8 here opens another tab (shouldn't it go to same window for consistency?)

![screen shot 2016-02-01 at 10 54 17 am](https://cloud.githubusercontent.com/assets/11320314/12727538/4b252b64-c8d2-11e5-94c1-54cfb6194814.png)

**Comment by mrmin123** [comment]
If you show the Individuals under a Family, you'd also be showing the Families and Individuals under a Group? What about Individuals under the Families of the Group?

I think that would be a separate ticket because the current breadcrumb system seems like a bad place to implement such hierarchal display (see Delete modal).

**Comment by mrmin123** [comment]
Link to internal resource from Variant curation page opening a new tab has been fixed.

**Comment by wrightmw** [comment]
I think it's better to have the briefcase linkable so that users get used to the functionality.

**Comment by wrightmw** [comment]
@mrmin123 You'll be pleased to know that I don't have any other issues :-)

**Comment by selinad** [comment]
Clicking on the Briefcase in this breadcrumb takes me to a page without the PMID selected (e.g. blank in the middle page) - odd as this is first I've observed this

![screen shot 2016-02-01 at 11 45 01 am](https://cloud.githubusercontent.com/assets/11320314/12728996/5cfe4e54-c8d9-11e5-8cef-79ab132b9c58.png)

**Comment by selinad** [comment]
uh - got this when I tried to add new experimental data - can't select type.  You can't see it in the screencapture, but hovering over gives that round circle with the slash through it..

![screen shot 2016-02-01 at 11 51 46 am](https://cloud.githubusercontent.com/assets/11320314/12729181/45b64e08-c8da-11e5-9235-05d5e1dac2e3.png)

**Comment by selinad** [comment]
Also when I edit and existing one 
![screen shot 2016-02-01 at 11 59 55 am](https://cloud.githubusercontent.com/assets/11320314/12729398/5d3bc7f0-c8db-11e5-89f9-01cb9f823bf7.png)

**Comment by wrightmw** [comment]
It's weird because I cannot reproduce the error that Selina is seeing:
![expo](https://cloud.githubusercontent.com/assets/15131169/12729506/f9724374-c8db-11e5-8fa8-df8efac9e3a2.png)

**Comment by selinad** [comment]
Just reproduced it but not sure as to exact series of events.  If I go out of the record and re-enter (actually, I brought up URL for instance again), it's fine.  But after a series of events testing the briefcase, I then get the problem again.  @wrightmw trying to reproduce again.  Seems like it has something to do with the briefcase navigation....

**Comment by wrightmw** [comment]
When I'm on the All Gene-Disease Records page, I cannot see that Selina has recently edited DICER1... it says date last edited of Feb 24th last year, see:
![history](https://cloud.githubusercontent.com/assets/15131169/12731227/bca92670-c8e4-11e5-9a2b-dbad8319dbc8.png)

**Comment by mrmin123** [comment]
@selinad I can't seem to reproduce that issue. If it's difficult to reproduce/is a super edge case, perhaps we can readdress it in the future?
@wrightmw Currently the 'Last Edited' field only accounts fo the last time the GDM object specifically was edited (this currently only happens when a PMID is added to the GDM).

**Comment by selinad** [comment]
@mrmin123 it's a pretty big problem as it prevents Experimental entry.  Let me see if I can figure out the sequence of events. It may be an edge case as it happened when testing the briefcase links, but I'm not sure yet...

**Comment by mrmin123** [comment]
@selinad @wrightmw This is a more up-to-date version of this branch with the latest `dev` merged in: http://326-mc-gdm-linkbacks-breadcrumbs-24aab99-minchoi.instance.clinicalgenome.org/

Please see if you can reproduce the error on this branch.

**Comment by selinad** [comment]
Thanks @mrmin123 it's reared its ugly head again. 

Here is what I did (after closing Chrome and reopening, according to @kilodalton 's suggestion, which I just now was able to test):

* made a new Experimental
* went to Edit it - only thing I could do was assess
* hit Cancel and went to make new Experimenta
* everything grayed out on new...

**Comment by wrightmw** [comment]
I really would like to support my fellow *ight in this but I have not been able to reproduce this error. However I have witnessed, first hand no less, it occurring twice for Selina. I also saw it happen once for Kang. It seems to be a sporadic problem.

**Comment by mrmin123** [comment]
@selinad 

* Did you add variant(s) and/or assessment, or did it not matter?
* Where did you go to edit from? After going to Curation Central and hitting the Edit button from the Curation Palette? By hitting back from the Submit page? By hitting Edit/Assess from the edit page?

**Comment by selinad** [comment]
HI @mrmin123 

* I did not add variants and I do not believe I assessed
* I believe I edited from the palette
* I was doing Bioch Function A

**Comment by selinad** [comment]
I think I have a clue - it seems to happen only after I edit what must have been test data (Experiment 8), so maybe all is OK?

<img width="257" alt="screen shot 2016-02-06 at 9 28 56 am" src="https://cloud.githubusercontent.com/assets/11320314/12868092/1d31cf88-ccb4-11e5-82db-67f7381ec2c2.png">

**Comment by mrmin123** [comment]
@selinad OK, thanks for narrowing that down. I was able to reproduce the error after changing ownership of the test data to myself.

Observations are as follows:

* Sometimes, viewing the 'View/Assess' page for Experiment 8 will not display the Assessment Panel. Refreshing the page brings it back.
* Upon entering the Edit page, only the Assessment can be changed (as it should be, as there are other assessments on the page)
* Upon changing the assessment, the user is forwarded to Curation Central
  * If the user directly goes into Add Experimental Data from here, the choose Type is grayed out. If the user refreshes, it becomes editable again.
  * If the user directly goes into Add Experimental Data from here, encounters the disabled Type field, navigates back to Curation Central, and goes into Add Experimental Data again, the Type field is still disabled
  * If the user refreshes Curation Central prior to going into Add Experimental Data, the Type is editable to begin with
  * If the user goes to Edit ANOTHER Experimental Data entry that is normally editable, it is also disabled, as if someone has already assessed on it, even if they haven't

From this I can only surmise that either the Assessment module or the Save button is making the variables that control the enabling/disabling of the form fields stick across pages. More testing is required to ascertain whether the issue is specific to this branch or is more widespread.

**Comment by selinad** [comment]
So far this is looking really good! (still testing a bit more).

Here is one strange thing, and it it also attached to Experiment 8 from what I can see - the ClnVar preferred title is displaying for variant  55847, but the other ClinVar variant I entered (furthest left) did not do that.  Not sure what is going on there and if it is just related to the testing data (or if I've missed something).
![screen shot 2016-02-06 at 1 28 31 pm](https://cloud.githubusercontent.com/assets/11320314/12869489/03322336-ccd6-11e5-966d-2a8ce190efa8.png)

**Comment by selinad** [comment]
Finally, someone can reproduce! ;-)  Thanks for working through the strange issue, @mrmin123

**Comment by selinad** [comment]
@mrmin123 I think there is something wrong with Assessment overall in this instance - I can't assess on a Family or new Experimental either...although Variant assessment seems unaffected.

**Comment by selinad** [comment]
I think the View/Assess is behaving like the View from the History pages...?

**Comment by mrmin123** [comment]
@selinad The variant not updating is because it is test (old) data. The ClinVar modal does not exist in this branch.

I'm not sure if the Assessment is the culprit yet, because the Assessment code wasn't touched in this branch...

**Comment by selinad** [comment]
Sorry @mrmin123 - a bit confused.  The variant that is updating is the old test data.  I didn't expect it to update.  The new variant is not updating, which I did expect.

**Comment by selinad** [comment]
Everything related to the return to GDM looks great!

**Comment by mrmin123** [comment]
Oh, I just looked at the test data. Somebody (probably @kgliu) updated the test data to have the Preferred Titles. Any new variants you add will not have this field because the Add ClinVar ID modal does not exist on this branch.

**Comment by selinad** [comment]
ok - that makes sense. Is this causing the bug since Experiment 8 is the odd man out in both cases (displaying ClinVar and the strange sticky bug)

**Comment by mrmin123** [comment]
@selinad I won't be around until later tonight to look at the code, but the only way to ascertain whether it's specific to Experiment 8 or not would be to create another Experimental Data with others assessing it to see if it can be reproduced elsewhere. I can do this later tonight if you don't get to it.

**Comment by selinad** [comment]
No hurry @mrmin123 

Here's the issue - here's a different Experimental - no ability to Assess.

![screen shot 2016-02-06 at 2 26 49 pm](https://cloud.githubusercontent.com/assets/11320314/12869797/67aa1d16-ccde-11e5-9293-994c8afc5c1d.png)

If I go to curation-test, this ability to Assess does exists.

However, Experiment 8 in test is also messed up wrt Edit (View/Assess is OK) in curation-test:
![screen shot 2016-02-06 at 2 34 03 pm](https://cloud.githubusercontent.com/assets/11320314/12869808/c68708b2-ccde-11e5-8887-7284a4a1acd3.png)

**Comment by selinad** [comment]
General summary: Experiment 8 seems to be causing problems and, maybe separately, the Ability to Assess F or Exp is  messed up in this instance....

**Comment by mrmin123** [comment]
@selinad I just confirmed that this bug occurs on other instances (such as http://curation-test.clinicalgenome.org/ ), as well as other data types (Families, specifically). The issue arises whenever you Edit the assessment of an object already assessed by another user, and then go to Add the same data type (with Add Family, you will get an error in the console saying that you cannot load the segregation, so you have to press the link multiple times to get it to go through).

I also noticed that the Family information is editable despite another user having assessed on it.

I know this is pretty late in the cycle, but I think this is for another ticket...

**Comment by kilodalton** [comment]
@selinad and @mrmin123 Thanks for tracking down the conditions to cause the problem. Since this preexisting bug is not specifically about navigation (i.e. this ticket), let's move this ticket along and evaluate on Monday whether or not the bug that has been can be fixed in this release. It will be in a separate ticket.

**Comment by selinad** [comment]
thanks @mrmin123 and @kilodalton 
@mrmin123 I'm thinking it might be better if you make the ticket as you have a better understanding of the problem?

**Comment by mrmin123** [comment]
Yes, I can take the ticket. Has one been created already?

**Comment by selinad** [comment]
Briefcase should be in front of VariationID instead of on its own line
![screen shot 2016-02-10 at 3 09 51 pm](https://cloud.githubusercontent.com/assets/11320314/12965352/5ef2ff4c-d00a-11e5-8524-4078d698b450.png)

**Comment by selinad** [comment]
This works well except the above ticket

**Comment by wrightmw** [comment]
Works well, except: When I'm curating data associated with a PMID that is at the bottom of the list then 
if I use the briefcase in an Individual, Family, Group or Experiment page it correctly take me back to the GDM with that PMID highlighted (e.g. Zaima et al)... 
![zaima](https://cloud.githubusercontent.com/assets/15131169/12965879/94a8b89e-d00e-11e5-9880-70ddf84bf364.png)


However if I click on the briefcase in a variant entry it takes me back to the GDM with a different PMID highlighted instead (e.g in this case it highlights Rauramo et al, instead of Zaima et al)
![rauramo](https://cloud.githubusercontent.com/assets/15131169/12965895/b031e68a-d00e-11e5-8f2f-c3790d1f7f90.png)
:
I think this is a #582 issue.

**Comment by kilodalton** [comment]
Included in last release (R4). Nice job and thanks for your hard work.
